lib/battle/BattleInfo.cpp

```diff
@@ -483,22 +483,44 @@ BattleInfo * BattleInfo::setupBattle(const int3 & tile, TerrainId terrain, const
 	//tactics
 	bool isTacticsAllowed = !creatureBank; //no tactics in creature banks
 
-	int tacticLvls[2] = {0};
-	for(int i = 0; i < ARRAY_COUNT(tacticLvls); i++)
+	std::array<std::pair<int,int>, 2> tacticLvls = {};

```

Comment:
Please use two arrays instead.
`battleRepositionLevel` and `battleRepositionLevelBlock` would be much more readable than `first` and `second`

Reply:
done